tests/cogs/test_information.py

```diff
@@ -0,0 +1,170 @@
+import asyncio
+import logging
+import textwrap
+from datetime import datetime
+from unittest.mock import MagicMock, patch
+
+import pytest
+from discord import (
+    CategoryChannel,
+    Colour,
+    Embed,
+    TextChannel,
+    VoiceChannel,
+)
+
+from bot.cogs import information
+from bot.constants import Emojis
+from bot.decorators import InChannelCheckFailure
+from tests.helpers import AsyncMock
+
+
+@pytest.fixture()
+def cog(simple_bot):
+    return information.Information(simple_bot)
+
+
+def role(name: str, id_: int):
+    r = MagicMock()
+    r.name = name
+    r.id = id_
+    r.mention = f'&{name}'
+    return r
+
+
+def member(status: str):
+    m = MagicMock()
+    m.status = status
+    return m
+
+
+@pytest.fixture()
+def ctx(moderator_role, simple_ctx):
+    simple_ctx.author.roles = [moderator_role]
+    simple_ctx.guild.created_at = datetime(2001, 1, 1)
+    simple_ctx.send = AsyncMock()
+    return simple_ctx
+
+
+def test_roles_info_command(cog, ctx):
+    everyone_role = MagicMock()
+    everyone_role.name = '@everyone'  # should be excluded in the output
+    ctx.author.roles.append(everyone_role)
+    ctx.guild.roles = ctx.author.roles
+
+    cog.roles_info.can_run = AsyncMock()
+    cog.roles_info.can_run.return_value = True
+
+    coroutine = cog.roles_info.callback(cog, ctx)
+    expected = Embed(
```

Comment:
Creating an embed object seems unnecessary if you're going to compare each field individually anyway.